osu.Game/Screens/Play/BeatmapMetadataDisplay.cs

```diff
@@ -167,6 +138,20 @@ private void load()
                 }
             };
 
+            metadataLine.AddText("Source", t =>
+            {
+                t.Colour = OsuColour.Gray(0.8f);
+                t.Padding = new MarginPadding { Right = 10 };
+            });
+            metadataLine.AddText(string.IsNullOrEmpty(metadata.Source) ? @"-" : metadata.Source);
+
+            metadataLine.AddParagraph("Mapper", t =>
+            {
+                t.Colour = OsuColour.Gray(0.8f);
+                t.Padding = new MarginPadding { Right = 10 };
+            });
+            metadataLine.AddText(string.IsNullOrEmpty(metadata.AuthorString) ? @"-" : metadata.AuthorString);
```

Comment:
This solution changes the design. Previously the "Source" and "Layout" texts were right-aligned with each other, and now they aren't.

From the screenshot you've provided:

![2020-10-04-024047_529x364_scrot](https://user-images.githubusercontent.com/20418176/95004423-1bfdbf80-05eb-11eb-89c3-87b1334a404f.png)


Additionally, as stated in the issue thread, we would probably be looking for enabling text wrapping wherever possible, but I guess that's a minor issue, relatively speaking.

Reply:
Sorry, based on dscnnctd's screenshot on 12 Nov 2019, I had assumed the design change was approved.
Just to clarify, which do you mean the first or second one?
![1](https://user-images.githubusercontent.com/5779211/95008082-61aa9000-05cb-11eb-98f2-9093a20c1008.png)
![2](https://user-images.githubusercontent.com/5779211/95008084-64a58080-05cb-11eb-9f87-daee782e855b.png)
Note that the second pic ran with a small hack prepending a space in front of Source so that it seems aligned.
I don't suppose that would get approved?

Regarding the max width and wrapping, what should the max width be?
I did a quick test to set a max width of 50 and it resulted in this which is clearly incorrect.
![3](https://user-images.githubusercontent.com/5779211/95008034-ed6fec80-05ca-11eb-972d-d90138cadb7e.png)
I think it's caused by the MultilineTextBuilder using the default_never_fixed_width_characters in SpriteText?

Reply:
Adding a space is indeed not correct. It relies on particular font and string used and will break with localisation.

As for the wrapping, there's a reference screenshot in the issue showing about how wide the text should be.

Make sure you're setting the max width of the actual fill flow - from what you have on the screenshot it seems to be that you might be incorrectly passing the with to sprite text creation parameters, although I can't say for sure without seeing the code.